src/meta/src/hummock/model/key_range.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp;
+
+use risingwave_hummock_sdk::VersionedComparator;
+use risingwave_pb::hummock::KeyRange;
+
+/// Implements a subset methods of `hummock_sdk::KeyRange` for prost `KeyRange`
+pub trait KeyRangeExt {
+    fn full_key_overlap(&self, other: &Self) -> bool;
+    fn full_key_extend(&mut self, other: &Self);
+    fn inf() -> Self;
+    fn new(left: Vec<u8>, right: Vec<u8>) -> Self;
+    fn cmp(&self, other: &Self) -> cmp::Ordering;
```

Comment:
`cmp` may be conflict with `Ord`? How about split it to `Ord`?

Reply:
> `cmp` may be conflict with `Ord`? How about split it to `Ord`?

You mean implement Ord trait? We cannot because prost::Keyrange is from extern crate.

Reply:
Ok, I get it.